src/js/test/js.cpp

```diff
@@ -7,58 +7,148 @@
 
 TEST_CASE("Check KV Map access")
 {
-  INFO("Application context");
-  {
-    // Governance tables are read-only
-    REQUIRE(
-      _check_kv_map_access(ccf::js::TxAccess::APP, "public:ccf.gov.table"));
-    // Public internal tables are read-only
-    REQUIRE(_check_kv_map_access(
-      ccf::js::TxAccess::APP, "public:ccf.internal.table"));
-    // Public applications tables are read-write
-    REQUIRE(!_check_kv_map_access(ccf::js::TxAccess::APP, "public:table"));
-    // Private applications tables are read-write
-    REQUIRE(!_check_kv_map_access(ccf::js::TxAccess::APP, "table"));
-  }
+  using namespace ccf::js;
+  constexpr auto public_internal_table_name = "public:ccf.internal.table";
+  constexpr auto private_internal_table_name = "ccf.internal.table";
+
+  constexpr auto public_gov_table_name = "public:ccf.gov.table";
+  constexpr auto private_gov_table_name = "ccf.gov.table";
 
-  INFO("Read-only governance context (ballot, resolve)");
+  constexpr auto public_app_table_name = "public:table";
+  constexpr auto private_app_table_name = "table";
   {
-    // Governance tables are read-only
-    REQUIRE(
-      _check_kv_map_access(ccf::js::TxAccess::GOV_RO, "public:ccf.gov.table"));
-    // Public internal tables are read-only
-    REQUIRE(_check_kv_map_access(
-      ccf::js::TxAccess::GOV_RO, "public:ccf.internal.table"));
-    // Public applications tables are read-only
-    REQUIRE(_check_kv_map_access(ccf::js::TxAccess::GOV_RO, "public:table"));
-    // Private applications tables are read-only
-    REQUIRE(_check_kv_map_access(ccf::js::TxAccess::GOV_RO, "table"));
+    INFO("In application context");
+    {
+      INFO("Public internal tables are read-only");
+      REQUIRE(
+        _check_kv_map_access(TxAccess::APP, public_internal_table_name) ==
+        MapAccessPermissions::READ_ONLY);
+    }
+
+    {
+      INFO("Private tables in internal namespace are read-only");
+      REQUIRE(
+        _check_kv_map_access(TxAccess::APP, private_internal_table_name) ==
+        MapAccessPermissions::READ_ONLY);
```

Comment:
That surely should be illegal?

Reply:
Yes - I've now made this change so accessing private governance or private internal tables from application code is `ILLEGAL`.